### PR TITLE
Allow using heavy atoms only in the FragmentChooser

### PR DESCRIPTION
**Pull request by ptosco**
This PR addresses #4754 by adding two `CleanupParameters`:
* `largestFragmentChooserUseAtomCount`: enables/disables atom count as `LargestFragmentChooser` criterion
* `largestFragmentChooserHeavyOnly`: count only heavy atoms
A `LargestFragmentChooser` constructor taking `CleanupParameters` as parameter was added to both C++ and Python, and relevant unit tests.

### CODE REVIEW

File: Code/GraphMol/MolStandardize/MolStandardize.h [hunk 1]
```diff
@@ -70,6 +70,12 @@ struct RDKIT_MOLSTANDARDIZE_EXPORT CleanupParameters {
   bool tautomerReassignStereo{
       true};  //! Whether enumerate() should call assignStereochemistry
               //! on all generated tautomers (defaults to true)
+  bool largestFragmentChooserUseAtomCount{
+      true};  //! Whether LargestFragmentChooser should use atom
+              //! count as criterion (defaults to true)
+  bool largestFragmentChooserHeavyOnly{
```
Comment (greglandrum) [inline Code/GraphMol/MolStandardize/MolStandardize.h:76]: consistent with the above suggestion:
```suggestion
  bool largestFragmentChooserCountHeavyAtomsOnly{
```

File: Code/GraphMol/MolStandardize/MolStandardize.h [hunk 2]
```diff
@@ -70,6 +70,12 @@ struct RDKIT_MOLSTANDARDIZE_EXPORT CleanupParameters {
   bool tautomerReassignStereo{
       true};  //! Whether enumerate() should call assignStereochemistry
               //! on all generated tautomers (defaults to true)
+  bool largestFragmentChooserUseAtomCount{
+      true};  //! Whether LargestFragmentChooser should use atom
+              //! count as criterion (defaults to true)
```
Comment (greglandrum) [inline Code/GraphMol/MolStandardize/MolStandardize.h:75]: maybe:
```suggestion
              //! count as one of the criteria (defaults to true)
```

File: Code/GraphMol/MolStandardize/Wrap/rdMolStandardize.cpp [hunk 3]
```diff
@@ -176,7 +176,17 @@ BOOST_PYTHON_MODULE(rdMolStandardize) {
           "tautomerReassignStereo",
           &RDKit::MolStandardize::CleanupParameters::tautomerReassignStereo,
           "call AssignStereochemistry on all generated tautomers "
-          "(defaults to True)");
+          "(defaults to True)")
+      .def_readwrite("largestFragmentChooserUseAtomCount",
+                     &RDKit::MolStandardize::CleanupParameters::
+                         largestFragmentChooserUseAtomCount,
+                     "Whether LargestFragmentChooser should use atom "
+                     "count as criterion (defaults to True)")
```
Comment (greglandrum) [inline Code/GraphMol/MolStandardize/Wrap/rdMolStandardize.cpp:184]:
```suggestion
                     "count as one of the criteria (defaults to True)")
```

Comment (greglandrum) [reply Code/GraphMol/MolStandardize/Wrap/rdMolStandardize.cpp:184]: BTW: nicely done to capitalize `True` here. :-)

File: Code/GraphMol/MolStandardize/Wrap/rdMolStandardize.cpp [hunk 4]
```diff
@@ -176,7 +176,17 @@ BOOST_PYTHON_MODULE(rdMolStandardize) {
...
+      .def_readwrite("largestFragmentChooserHeavyOnly",
+                     &RDKit::MolStandardize::CleanupParameters::
+                         largestFragmentChooserHeavyOnly,
```
Comment (greglandrum) [inline Code/GraphMol/MolStandardize/Wrap/rdMolStandardize.cpp:187]:
```suggestion
                         largestFragmentChooserCountHeavyAtomsOnly,
```

**Review by greglandrum — CHANGES_REQUESTED** [review]
suggested naming changes

**Comment by ptosco** [comment]
@greglandrum Done; I have slightly reworded the comments but still along the lines of your suggestion (i.e., improved clarity).

**Review by greglandrum — APPROVED** [review]

**Pull request merged**